### DISCUSSION TITLE
CDT / CMSIS installed but not working properly

### DISCUSSION THREAD

**Discussion by alricht** [eclipse-embed-cdt/eclipse-plugins#542] [General Discussions]
Hi,
I still got some problems with my newly setup eclipse. It seems like some parts of eclipse are not aware of the Embedded CDT package / CMSIS / xPack. 
I'm not able to do a device selection when I go into the Properties of a current project. ("No devices available, install xPacks or CMSIS Pack first"). Both xPacks and CMSIS should be installed. The CMSIS Packs view is available (even though only the device list as the project isn't a CMSIS project) and Toolchain recognizes xPacks.
I'm also not able to create a new CMSIS project.  (as described https://github.com/ARM-software/cmsis-pack-eclipse)

Am I overseeing something, do I still have to activate something? Is the installation of Embedded CDT maybe corrupted on my system? The last time I did an embedded eclipse setup it was on windows. I didn't have these issues there. 

I'm using:
- eclipse Version: 2022-06 (4.24.0)
- Embedded CDT 6.3.0.202208180721
- Eclipse C/C++ IDE CDT 10.7 (2022-06)
- System VMware Ubuntu 22.04.1-desktop-amd64, Host System Windows 10

**Comment by ilg-ul** [comment]
> I'm also not able to create a new CMSIS project. (as described https://github.com/ARM-software/cmsis-pack-eclipse)

The link you mention refers to a different plug-in, with different functionality, and not integrated into Embedded CDT.

**Reply by alricht** [reply]
well ok I just thought the project generation looks the same as visualized in the link. I can delete the link if it's misleading.
I indeed have the following installed:
Embedded C/C++ CMSIS Packs (Expermimental), Eclipse Embedded CDT, 6.3.0.202208180721

**Comment by TommyMurphyTM1234** [comment]
You might want to explain the precise steps that you took to install the various bits and pieces?

**Comment by ilg-ul** [comment]
I also suggest  to follow the steps in: 

- <https://eclipse-embed-cdt.github.io/plugins/packs-manager/>
- <https://eclipse-embed-cdt.github.io/tutorials/blinky-arm/> 
 
to get a functional debug session using the CMSIS Packs SVDs, that you can use as reference for your project.

**Reply by eraus** [reply]
I was following the "blinky-arm" tutorial with the installation of all "latest" tools. However, there are build errors as shown below:

"""
Building file: ../system/src/newlib/syscalls.c
Invoking: GNU Arm Cross C Compiler
arm-none-eabi-gcc -mcpu=cortex-m4 -mthumb -mfloat-abi=soft -Og -fmessage-length=0 -fsigned-char -ffunction-sections -fdata-sections -fno-move-loop-invariants -Wall -Wextra -g3 -DDEBUG -DUSE_FULL_ASSERT -DOS_USE_SEMIHOSTING -DTRACE -DOS_USE_TRACE_SEMIHOSTING_DEBUG -DSTM32F407xx -DUSE_HAL_DRIVER -DHSE_VALUE=8000000 -I"../include" -I"../system/include" -I"../system/include/cmsis" -I"../system/include/stm32f4-hal" -std=gnu11 -MMD -MP -MF"system/src/newlib/syscalls.d" -MT"system/src/newlib/syscalls.o" -c -o "system/src/newlib/syscalls.o" "../system/src/newlib/syscalls.c"
../system/src/newlib/syscalls.c: In function 'findslot':
../system/src/newlib/syscalls.c:599:26: error: 'struct _reent' has no member named '__sdidinit'
  599 |       if ((ptr) && !(ptr)->__sdidinit)          \
      |                          ^~
../system/src/newlib/syscalls.c:613:3: note: in expansion of macro 'CHECK_INIT'
  613 |   CHECK_INIT(_REENT);
      |   ^~~~~~~~~~
make: *** [system/src/newlib/subdir.mk:39: system/src/newlib/syscalls.o] Error 1
"make all" terminated with exit code 2. Build might be incomplete.

21:05:10 Build Failed. 2 errors, 0 warnings. (took 4s.805ms)
"""

I assume this may be related to the incompatible versions of the tools. How to handle this?

**Reply by ilg-ul** [reply]
I confirm, with arm-none-eabi-gcc 11.3 I also get this error. :-(

Please retry with 11.2.

**Reply by ilg-ul** [reply]
In the new versions of newlib, the macro should read `(ptr)->__cleanup)`. I'll fix the template.

**Reply by eraus** [reply]
Ok, tried v11.2. Both the C and C++ versions work well. Many thanks!

**Reply by ilg-ul** [reply]
I already fixed the template to work on all toolchain versions, it'll be available in 6.3.1.

**Reply by ilg-ul** [reply]
https://github.com/eclipse-embed-cdt/eclipse-plugins/issues/543
